### PR TITLE
Add support for GA code

### LINKED ISSUE #323: [Analytics] Allow merchants to enable GA for their store
Allow merchants to go to settings > advanced and enable Google Analytics for their store by pasting in code from their Google Analytics account. They should still be able to view the reporting dashboards that we provide.

See [on Zeplin](https://zpl.io/bLM4mn6).

### PR DESCRIPTION
**Pull request by shahthepro**
Closes #323

### CODE REVIEW

File: backend/utils/deployShop.js [hunk 1]
```diff
@@ -196,13 +196,19 @@ async function _deployShop({
...
+        .replace('TITLE', publicShopConfig.fullTitle)
+        .replace('META_DESC', publicShopConfig.metaDescription || '')
+        .replace('DATA_DIR', dataDir)
+        .replace(/NETWORK/g, networkName)
+        .replace('FAVICON', publicShopConfig.favicon || 'favicon.ico')
+        .replace('UI_SRC', networkConfig.uiCdn || '')
+        // Add GA code just before </head> tag
+        // TODO: Technically any script can be injected,
+        // make sure to add a warning in frontend
+        .replace('</head>', `${publicShopConfig.gaCode || ''}</head>`)
```
Comment (shahthepro) [inline backend/utils/deployShop.js:210]: @nick @gullible1 Do we want to do add a warning to the frontend? Could open up XSS vulnerabilities if user pastes scripts from suspicious sites

Comment (nick) [reply backend/utils/deployShop.js:210]: I'm thinking we use the same mechanism as the Admin for GA. `See shop/src/pages/App.js` line 92.

Comment (shahthepro) [reply backend/utils/deployShop.js]: Have made that change, thanks for pointing it out

**Comment by shahthepro** [comment]
Merging this now, let me know if there any more comments, will address them

**Pull request merged**